torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -271,11 +283,28 @@ static void UpdateScalarTypeForOutput(
       CreateProfiledTensorTypeWithScalarType(output_tensor_type, scalar_type));
 }
 
+static void RecoverScalarTypeForOutput(
+    Node* n,
+    const c10::ScalarType& scalar_type) {
+  const int64_t onnx_type = ScalarTypeToONNXType(scalar_type);
+  Node* cast_node = n->owningGraph()->create(onnx::Cast, 1);
+  cast_node->addInput(n->output());
+  cast_node->i_(attr::to, onnx_type);
+  cast_node->insertAfter(n);
+  n->output()->replaceAllUsesAfterNodeWith(cast_node, cast_node->output());
+}
+
 static void ImplicitCastNodeForONNX(Node* n) {
   if (IsImplicitCastSupported(n->kind())) {
     auto expected_scalar_type = InferExpectedScalarType(n);
     if (expected_scalar_type) {
-      UpdateScalarTypeForInputs(n, *expected_scalar_type);
+      auto expected_scalar_type_cast =
+          LowPrecisionCastForStandardOps(n, *expected_scalar_type);
```

Comment:
Why can't you separate `LowPrecision` logic from `ImplicitCast`?

Reply:
Great point, fixed this now.